adapter/integration-test/src/adapter-datasource.test.js

```diff
@@ -98,6 +98,18 @@ describe('Datasource Configuration', () => {
     expect(datasourceResponse.status).toEqual(400)
   }, TIMEOUT)
 
+  test('Should create datasource with long URL [POST /datasources]', async () => {
+    const longUrlDatasourceConfig = getDatasourceConfig()
+    const queryParameter = '&parameter=longParameter'
+    const longUrl = 'http://www.verylongdomain.com?first=test' + queryParameter.repeat(20)
+    longUrlDatasourceConfig.protocol.parameters.location = longUrl
+    const response = await request(ADAPTER_URL)
+      .post('/datasources')
+      .send(longUrlDatasourceConfig)
+
+    expect(response.status).toEqual(201)
+  })
+
```

Comment:
What happens if someone attempts to POST a datasource with an URL longer than 2000 characters? Does the service crash or does it respond with HTTP 500? 
Maybe we should add another test to ensure the correct behavior.

The cleanest solution would be to check the incoming request in the API class and respond with HTTP 400 if fields are too long.
However, since this probably applies to more fields, it should probably be done in another PR.

Reply:
> What happens if someone attempts to POST a datasource with an URL longer than 2000 characters? Does the service crash or does it respond with HTTP 500?
> Maybe we should add another test to ensure the correct behavior.

With the change to support unlimited length this is no longer an issue.

> However, since this probably applies to more fields, it should be done in another PR.

I agree we should add more validation logic in another pull request.